### ISSUE TITLE
Does it possible to lint React components in mdx files?

### ISSUE THREAD

**Issue by B2o5T** [mdx-js/eslint-mdx#388]
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Problem

For example I have a mdx file
```jsx
// example.mdx

<Example prop={"foo"}></Example>
```

### Solution

with configured `react/self-closing-comp` and `react/jsx-curly-brace-presence` should suggest to replace it to
```jsx
// example.mdx

<Example prop="foo" />
```

**Comment by JounQin** [comment]
Please provide a minimal and runnable reproduction.

**Comment by B2o5T** [comment]
It's possible, at first I was confused `overrides` configuration.

**Comment by B2o5T** [comment]
Btw you should keep the line order of errors, note that errors from `mdx` file come first than the error from `js` code block, but I expected to see first errors from line 4 and after from line 7 and not viceverca

```mdx
# Hello world

\```js
<Example prop={"foo"}></Example>
\```

<Example prop={"foo"}></Example>
```

```
/Users/dimitri/Desktop/mdx-repro/test.mdx
  7:1   error  Empty components are self-closing  react/self-closing-comp
  7:15  error  Curly braces are unnecessary here  react/jsx-curly-brace-presence
  4:1   error  Empty components are self-closing  react/self-closing-comp
  4:15  error  Curly braces are unnecessary here  react/jsx-curly-brace-presence

✖ 4 problems (4 errors, 0 warnings)
  4 errors and 0 warnings potentially fixable with the `--fix` option.
```

**Comment by JounQin** [comment]
The order seems making sense, PR welcome to reorder them correctly at https://github.com/mdx-js/eslint-mdx/blob/fd18bed8374e23d64128481c9b17afefee5164ba/packages/eslint-plugin-mdx/src/processors/remark.ts#L29
